### PR TITLE
Implement Enum Validation

### LINKED ISSUE #44: Validating against enum
Is there a way to validate an argument to ensure that it has only a list of specific values?

For example

```
func(
          suit = Json(hearts | clubs | spades | diamonds)
):
```

Or maybe validate against an existing Python enum

### PR DESCRIPTION
**Pull request by smt5541**
#### Give examples of the changes you've made in this pull request. Include an itemized list if you can.

- Add validation for `StrEnum` and `IntEnum` subclasses.

#### Why did you choose to make these changes?

- I have frequently used the `func` parameter to validate my `Enum` parameters from `str` or `int`, but this is overhead that could easily be included in this library to reduce many similar validation functions

#### Does this pull request resolve any open issues?

Yes. Closes #44 

#### Were there any trade-offs you had to consider?

Not particularly

### 🧪 Testing

- [X] Have tests been added or updated for the changes introduced in this pull request?

- [X] Are the changes backwards compatible?

#### If the changes aren't backwards compatible, what other options were explored?

N/A

### ✨ Quality check

- [X] Are your changes free of any erroneous print statements, debuggers or other leftover code?

- [X] Has the README been updated to reflect the changes introduced (if applicable)?

#### Feel free to add any further information below

Dependent on #45

### CODE REVIEW

**Comment by Ge0rg3** [comment]
Hey, I'm a little hesitant on the StrEnum import as it'll lock us into Py3.11+ when this was added.

Perhaps for Strs we could just use typing.Literal instead? Eg
```py
@app.route("/fruitshop/buy", methods=["POST"])
@ValidateParameters()
def buy_fruit(
    fruit_name: typing.Literal["apple", "banana", "cherry"] = Json(),
):
    return "Fruit bought!"
```

File: flask_parameter_validation/parameter_types/parameter.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 """
 import re
 from datetime import date, datetime, time
+from enum import Enum, StrEnum, IntEnum
```
Comment (Ge0rg3) [inline flask_parameter_validation/parameter_types/parameter.py:7]: Risky StrEnum as per comment

**Review by Ge0rg3 — CHANGES_REQUESTED** [review]

File: flask_parameter_validation/parameter_types/parameter.py (hunk 1 above)
Comment (smt5541) [reply flask_parameter_validation/parameter_types/parameter.py:7]: What do you think about performing the import, catching `ImportError` if it is raised, on which we fall back on checking for `issubclass(allowed_types[0], Enum) and issubclass(allowed_types[0], str)` to replace the `StrEnum` import, and `issubclass(allowed_types[0], Enum) and issubclass(allowed_types[0], int)` to replace the `IntEnum` import?

Comment (Ge0rg3) [reply flask_parameter_validation/parameter_types/parameter.py:7]: Sounds like a great solution, especially if also compatible with StrEnum anyway :D

Comment (smt5541) [reply flask_parameter_validation/parameter_types/parameter.py]: Implemented ✅

**Comment by smt5541** [comment]
Hey @Ge0rg3 just thought I should bump this, no rush though!

**Comment by Ge0rg3** [comment]
Hey @smt5541, please can you show some sample usage for this? Thanks!

**Review by Ge0rg3 — APPROVED** [review]

**Comment by Ge0rg3** [comment]
Thanks!

**Pull request merged**